test/boost/s3_test.cc

```diff
@@ -377,3 +378,33 @@ SEASTAR_THREAD_TEST_CASE(test_client_put_get_tagging) {
         BOOST_CHECK(tagset.empty());
     }
 }
+
+SEASTAR_THREAD_TEST_CASE(test_client_list_objects) {
+    const sstring bucket = tests::getenv_safe("S3_BUCKET_FOR_TEST");
+    const sstring prefix(fmt::format("testprefix-{}/", ::getpid()));
+    semaphore mem(16<<20);
+    auto client = s3::client::make(tests::getenv_safe("S3_SERVER_ADDRESS_FOR_TEST"), make_minio_config(), mem);
+    auto close_client = deferred_close(*client);
+    std::unordered_set<sstring> names;
+
+    // Put extra object to check list-by-prefix filters it out
+    temporary_buffer<char> data = sstring("1234567890").release();
+    client->put_object(format("/{}/extra-{}", bucket, ::getpid()), std::move(data)).get();
+    for (int i = 0; i < 12; i++) {
+        temporary_buffer<char> data = sstring("1234567890").release();
+        auto name = format("{}obj.{}", prefix, i);
+        client->put_object(format("/{}/{}", bucket, name), std::move(data)).get();
```

Comment:
`data` has been moved away when putting the extra object and when putting the previous object with `f"testprefix-{pid}"` prefix. so we are actually putting 12 empty objects with the specified prefix.  to minimize surprise, probably we could put a duplicated payload? like:
```c++
client->put_object(format("/{}/extra-{}", bucket, ::getpid()), data.share()).get();
// ...
client->put_object(format("/{}/{}", bucket, name), data.share()).get();
```

or, in this case, we don't really care about the content of the objects, we could just use an empty payload.

Reply:
> so we are actually putting 12 empty objects with the specified prefix.

Why? Inside this loop there's another variable that shadows the outer one, no?

> or, in this case, we don't really care about the content of the objects, we could just use an empty payload.

Yes, we don't care the content

Reply:
> > so we are actually putting 12 empty objects with the specified prefix.
> 
> Why? Inside this loop there's another variable that shadows the outer one, no?

ahh, indeed. i didn't realize that you are reusing the same variable name. sorry for the noise.